source/beamline/BioXAS/BioXASZebraLogicBlock.h

```diff
@@ -0,0 +1,66 @@
+#ifndef BIOXASZEBRALOGICBLOCK_H
+#define BIOXASZEBRALOGICBLOCK_H
+
+#include "beamline/AMControl.h"
+#include "beamline/AMPVControl.h"
+#include "beamline/AMControlSet.h"
+
+#include "beamline/BioXAS/BioXASZebraLogicBlockInput.h"
+
+#define BIOXASZEBRALOGICBLOCK_INPUT_NUM 4
+
+class BioXASZebraLogicBlock : public AMControl
+{
+    Q_OBJECT
+
+public:
+	/// Enumeration of the possible output state values.
+	enum State { Low = 0, High = 1 };
+
+	/// Constructor.
+	explicit BioXASZebraLogicBlock(const QString &name, const QString &baseName, QObject *parent = 0);
+	/// Destructor.
+	virtual ~BioXASZebraLogicBlock();
+
+	/// Returns the connected state.
+	bool isConnected() const { return connected_; }
+
+	/// Returns the output state value.
+	double outputStateValue() const;
+	/// Returns true if the output state is low.
+	bool isStateLow() const { return !isStateHigh(); }
+	/// Returns true if the output state is high.
+	bool isStateHigh() const;
```

Comment:
It would probably be better to be explicit with these names since there are many states.  (`bool isOutputStateHigh() const;`)